test/model/store_test.dart

```diff
@@ -391,6 +392,22 @@ void main() {
       check(store.userSettings!.twentyFourHourTime).isTrue();
     }));
 
+    String? lastReportedError;
+    String? takeLastReportedError() {
+      final result = lastReportedError;
+      lastReportedError = null;
+      return result;
+    }
+
+    void logAndReportErrorToUserBriefly(String? message, {String? details}) {
+      if (message == null) return;
+      lastReportedError = '$message\n$details';
+    }
+    // This overrides the function globally, but we don't need to worry about
+    // leaking the state because that is only relevant to widget tests.
+    // See `lib/log.dart ` for details.
+    reportErrorToUserBriefly = logAndReportErrorToUserBriefly;
```

Comment:
Better to just systematically avoid leaking state between tests.

In general the way to do that is to (a) set it up from within a test case, and then (b) use `addTearDown` to ensure it gets reset when the test case is done. Given the function definition above, I think code like
```dart
  test('some test', () {
    reportErrorToUserBriefly = logAndReportErrorToUserBriefly;
    addTearDown(reportErrorToUserBriefly = defaultReportErrorToUserBriefly);
    // … rest of test …
```

should do it.

When the setup is more complicated than that, that's the sort of thing that we write helper functions named like "prepareFoo" for.